### PR TITLE
Fieldset generation feature

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: starter-core/src/test/groovy/io/micronaut/starter/feature/test/JunitPlatformSuiteEngineSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        where:
+        applicationType << ApplicationType.values()
+    }
+
+    @Unroll
+    void 'test #buildTool junit-platform-suite-engine feature for language=#language'(Language language, BuildTool buildTool) {
+        when:
+        String template = new BuildBuilder(beanContext, buildTool)
+                .language(language)
+                .testFramework(TestFramework.JUNIT)
```
Comment (timyates) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/test/JunitPlatformSuiteEngineSpec.groovy:42]: Should we test with Spock as well?

There's no limitation on testFramework is there?  Not sure if it makes sense for Kotest 🤔

File: starter-core/src/main/java/io/micronaut/starter/feature/view/Thymeleaf.java [hunk 2]
```diff
@@ -17,12 +17,44 @@
...
+     * @deprecated Use {@link Thymeleaf(ViewsFieldset)} instead.
+     */
+    @Deprecated(forRemoval = true, since = "4.2.0")
+    public Thymeleaf() {
+        this(new ViewsFieldset());
+    }
+
+    @Override
+    public void processSelectedFeatures(FeatureContext featureContext) {
+        featureContext.addFeatureIfNotPresent(ViewsFieldset.class, viewsFieldset);
```
Comment (timyates) [inline starter-core/src/main/java/io/micronaut/starter/feature/view/Thymeleaf.java:55]: So Thymeleaf will always add ViewsFieldset?

**Review by timyates — APPROVED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/view/Thymeleaf.java (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/view/Thymeleaf.java:55]: yes.

File: starter-core/src/test/groovy/io/micronaut/starter/feature/test/JunitPlatformSuiteEngineSpec.groovy (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/test/JunitPlatformSuiteEngineSpec.groovy:42]: I think it is fine.

**Pull request merged**